### PR TITLE
JABERA-5 added Post files with code

### PR DESCRIPTION
**Pull request by jkk84**

### CODE REVIEW

File: src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    private final PostService postService;
+
+    @GetMapping
+    public List<PostDto> getAllPosts(@RequestParam long userId) {
+        return postService.getAllPosts(userId);
+    }
+
+    @PostMapping
+    public ResponseEntity<PostEntity> addPost(@RequestBody PostDto postDto) {
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java]: Let's return PostDto

File: src/main/java/com/the/dark/side/crew/fejsbuk/mapper/PostMapper.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    private final UserRepository userRepository;
+
+    public List<PostDto> toDtos(List<PostEntity> allPosts) {
+        return allPosts.stream()
+                .map(this::toDto)
+                .collect(Collectors.toList());
+    }
+
+    public PostDto toDto(PostEntity postEntity) {
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/mapper/PostMapper.java:27]: Create separate class for mapping methods

File: src/main/java/com/the/dark/side/crew/fejsbuk/mapper/PostMapper.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        return PostDto.builder()
+                .id(postEntity.getId())
+                .message(postEntity.getMessage())
+                .localDateTime(postEntity.getCreatedAt())
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/mapper/PostMapper.java]: rename postDto localDateTime field

File: src/main/java/com/the/dark/side/crew/fejsbuk/model/PostEntity.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Setter;
+import org.hibernate.annotations.CreationTimestamp;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/model/PostEntity.java]: This annotation has no effect if none constructor was specified

File: src/main/java/com/the/dark/side/crew/fejsbuk/service/PostService.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.the.dark.side.crew.fejsbuk.service;
+
+import com.the.dark.side.crew.fejsbuk.model.PostEntity;
+import com.the.dark.side.crew.fejsbuk.model.dto.PostDto;
+
+import java.util.List;
+
+public interface PostService {
+
+    List<PostDto> getAllPosts(long id);
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/service/PostService.java]: rename parameter to userId

**Review by rmaj91 — COMMENTED** [review]
Few comments. But still test needed

File: src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java [hunk 6]
```diff
@@ -15,15 +15,16 @@
 @RequestMapping("/posts")
 public class PostController {
 
-    private final PostService postService;
+    private final PostServiceImpl postService;
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java]: We use interfaces when injecting beans

File: src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java [hunk 7]
```diff
@@ -15,15 +15,16 @@
...
     @GetMapping
     public List<PostDto> getAllPosts(@RequestParam long userId) {
         return postService.getAllPosts(userId);
     }
 
     @PostMapping
-    public ResponseEntity<PostEntity> addPost(@RequestBody PostDto postDto) {
-        return ResponseEntity.ok(postService.addPost(postDto));
+    public ResponseEntity<PostDto> addPost(@RequestBody PostDto postDto) {
+        return ResponseEntity.ok(postMapper.toDto(postService.addPost(postDto)));
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java]: Controller it's not good place for mapping, do it in service

File: src/main/java/com/the/dark/side/crew/fejsbuk/model/UserEntity.java [hunk 8]
```diff
@@ -17,13 +17,18 @@
...
     private String firstName;
 
-    @Column(name = "last_name")
     private String lastName;
 
     private String login;
 
     private String email;
+
+    public UserEntity(String firstName, String lastName, String login, String email) {
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/model/UserEntity.java]: If you're not using it, just remove and get rid of @NoArgsContractor. Same with postEntity

**Comment by rmaj91** [comment]
.

**Review by rmaj91 — COMMENTED** [review]
.

**Review by rmaj91 — COMMENTED** [review]
Tests left

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java [hunk 9]
```diff
@@ -2,72 +2,67 @@
...
-//
-////        when(postRepository.findByUserEntityId(userId)).thenReturn(List.of(postEntity));
-//        when(postMapper.toDtos(postRepository.findByUserEntityId(userId))).thenReturn(List.of(postDto));
-//
-//        List<PostDto> result = postService.getAllPosts(userId);
-//        assertTrue(result.);
-//        assertArrayEquals(result, postDto);
-//    }
+    @Test
+    void whenUserFoundThanReturnAllPosts() {
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: *Then*

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java [hunk 10]
```diff
@@ -2,72 +2,67 @@
...
+        long userId = 1L;
+        PostDto postDto = mock(PostDto.class);
+
+        when(postMapper.toDtos(postRepository.findByUserEntityId(userId))).thenReturn(List.of(postDto));
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: Its actually not testing anything, you're passing null

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java [hunk 11]
```diff
@@ -2,72 +2,67 @@
...
+        List<PostDto> result = postService.getAllPosts(userId);
+        assertEquals(1, result.size());
+        assertEquals(postDto, result.get(0));
+    }
+
+    @Test
+    void whenUserNotFoundThanReturnEmptyList() {
+        long userId = 1L;
+
+        when(postMapper.toDtos(postRepository.findByUserEntityId(userId))).thenReturn(List.of());
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: Same as previously. And use Collections.emptyList();

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java [hunk 12]
```diff
@@ -2,72 +2,67 @@
...
-        // then
-//        PostDto post = postMapper.toDto(postService.addPost(postDto));
-//        post.setUserId(1);
-//        verify(postRepository).save(postMapper.toEntity(postDto));
-//        assertEquals(post, postDto);
+        PostDto postDto = mock(PostDto.class);
+        PostEntity postEntity = mock(PostEntity.class);
+
+        when(postMapper.toEntity(postDto)).thenReturn(postEntity);
+        when(postRepository.save(postEntity)).thenReturn(postEntity);
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: not necessary, and use verify on .save method

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/UserServiceTest.java [hunk 13]
```diff
@@ -28,29 +28,33 @@ void init() {
...
         when(userRepository.findById(userId)).thenReturn(Optional.of(userEntity));
 
-        //then
         Optional<UserEntity> result = userService.getUser(userId);
         assertTrue(result.isPresent());
         assertEquals(userEntity, result.get());
     }
 
+    @Test
+    void whenUserIdNotFindThenReturn404() {
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/UserServiceTest.java]: method should have name: whenUserNotFoundThenEmpty

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/UserServiceTest.java [hunk 14]
```diff
@@ -28,29 +28,33 @@ void init() {
     }
 
     @Test
-    void shouldReturnUser() {
-        // given
+    void whenUserIdFoundThenReturnUser() {
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/UserServiceTest.java]: Your not searching userId but user

**Review by rmaj91 — COMMENTED** [review]
:dancers:

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java (hunk 12 above)
Comment (jkk84) [reply src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: when i remove this test fail

File: src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+public class PostController {
+
+    private final PostService postService;
+
+    @GetMapping
+    public List<PostDto> getAllUserIdPosts(@RequestParam long userId) {
+        return postService.getAllUserIdPosts(userId);
+    }
+
+    @GetMapping("/all")
```
Comment (rmaj91) [inline src/main/java/com/the/dark/side/crew/fejsbuk/controller/PostController.java]: It's not good path name

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+    void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void whenUserFoundThenReturnAllPosts() {
+        long userId = 1L;
+        PostDto postDto = mock(PostDto.class);
+
+        when(postMapper.toDtos(postRepository.findByUserEntityId(userId)))
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: its resolve to: when(postMapper.toDtos(null))
So it's not good test

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+        List<PostDto> result = postService.getAllUserIdPosts(userId);
+        assertEquals(1, result.size());
+        assertEquals(postDto, result.get(0));
+    }
+
+    @Test
+    void whenUserNotFoundThenReturnEmptyList() {
+        long userId = 1L;
+
+        when(postMapper.toDtos(postRepository.findByUserEntityId(userId))).thenReturn(Collections.emptyList());
```
Comment (rmaj91) [inline src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: same

File: src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java (hunk 12 above)
Comment (rmaj91) [reply src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/PostServiceTest.java]: but it should, u're never using returned refference from this method

**Review by rmaj91 — COMMENTED** [review]
.

**Review by rmaj91 — APPROVED** [review]
OK

**Pull request merged**